### PR TITLE
feat: Present Proof - adds command support

### LINKED ISSUE #1633: Present Proof - need to add command support
Need to add `command` support.

### PR DESCRIPTION
**Pull request by soluchok**
Adds `command` support for the Present Proof protocol.

Closes #1633 
Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/controller/command/presentproof/command.go [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+
+	// creates action channel
+	actions := make(chan service.DIDCommAction)
+	// registers action channel to listen for events
+	if err := client.RegisterActionEvent(actions); err != nil {
+		return nil, fmt.Errorf("register action event: %w", err)
+	}
+
+	// this code listens for the action events and does nothing
+	// this trick is used to avoid error such as `no clients are registered to handle the message`
```
Comment (troyronda) [inline pkg/controller/command/presentproof/command.go:99]: @rolsonquadras @llorllale

Comment (llorllale) [reply pkg/controller/command/presentproof/command.go:99]: Is this an architecture design question? Should protocol services require registered listeners for `DIDCommAction` events when handling inbound messages?

Comment (soluchok) [reply pkg/controller/command/presentproof/command.go:99]: now, I am having trouble with that design in my next PR. I suggest removing that restriction `no clients are registered to handle the message`.  We do not need a client to be registered bc we can continue at any time by using rest. @troyronda @llorllale Do you agree?

Comment (soluchok) [reply pkg/controller/command/presentproof/command.go:99]: This trick doesn't work anyway.  I forgot that we can register only one channel.

Comment (soluchok) [reply pkg/controller/command/presentproof/command.go:99]: Or is it the idea? I mean that only one consumer can be registered.

File: pkg/controller/command/presentproof/models.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+import "github.com/hyperledger/aries-framework-go/pkg/client/presentproof"
+
+// DeclinePresentationArgs model
+//
+// This is used when the presentation needs to be rejected
+//
+type DeclinePresentationArgs struct {
+	// PIID protocol state machine identifier
+	PIID string `json:"piid"`
```
Comment (Baha-sk) [inline pkg/controller/command/presentproof/models.go:17]: `omitempty` is not needed here and all json definitions below?

Comment (soluchok) [reply pkg/controller/command/presentproof/models.go:17]: No, we do not need `omitempty`. I need to Unmarshal incoming data/arguments, but I do not need to Marshal them.

File: pkg/controller/command/presentproof/command.go (hunk 1 above)
Comment (rolsonquadras) [reply pkg/controller/command/presentproof/command.go:99]: Note - DIDExchange sends action event only when clients are registered. https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/didexchange/service.go#L360

**Pull request merged**